test/format/format_skip_errors.sh

```diff
@@ -0,0 +1,30 @@
+#! /bin/bash
+
+# Conditional grep
+grep -q "cache clean check: no" RUNDIR.*.log && grep -q "cache dirty check: no" RUNDIR.*.log && grep "format run more than 15 minutes past the maximum time" RUNDIR.*.log
+
+if [ $? -eq 0 ]
+then
+	echo "{cache clean check: no} && {cache dirty check: no} && {format run more than 15 minutes past the maximum time} Error found in the failed log"
+	exit 0
+fi
+
+# skip_list : Append the errors to be skipped
+declare -a skip_list=(  "current_state == WT_REF_LIMBO || current_state == WT_REF_MEM"
+                        "heap-buffer-overflow"
+                     )
+
+# Iterate the skip error list and search in the failed logs.
+# Exit with status 0 if found, else exit with status 1
```

Comment:
The criterion of returning 0 is actually any matching of the given list of known errors, which could be problematic when both a known error and an unknown error are captured by the test (task run). The unknown error will be overshadowed by the known error which is undesired. 

This makes me think of the best place of introducing this "skipping known errors" logic is actually the format.sh script. Specifically, the place where a job result is set to fail in the script.

Reply:
I agree the ideal solution would be to include the "skip known errors" logic in the format.sh script and to skip errors after each failure.